### ISSUE TITLE
Plugin Host Fields should be separated

### ISSUE THREAD

**Issue by netniV** [Cacti/cacti#3756]
# Feature Request

## Is your feature request related to a problem? Please describe

In order to maintain clearer table ownership and comparision/stability, tables such as the `host` tab should not be modified by plugins.

## Describe the solution you'd like

Warnings if the table is modified from the expected structures.  This should probably be done using the Cacti Audit program during a weekly maintenance cycle.  

However, to facilitate this, we would need to call plugins during field substitution to afford them the opportunity to replace fields we had not.  There should also be a common method of doing this as the null_out_hostfields also replaces these

**Comment by TheWitness** [comment]
I don't think this is so much a Cacti problem, but a Developer problem.  I for one have made this mistake multiple times.  I have a few plugins that are like this.  So, if we ever get into the 'certification' business, we should handle that there.  It's not really a bug that I even want to spend time on.  It is an annoyance though.
